libs/testserver/fake_workspace.go

```diff
@@ -157,6 +157,27 @@ func (s *FakeWorkspace) JobsCreate(request jobs.CreateJob) Response {
 	}
 }
 
+func (s *FakeWorkspace) JobsReset(request jobs.ResetJob) Response {
+	jobId := request.JobId
+
+	_, ok := s.jobs[request.JobId]
+	if !ok {
+		return Response{
+			StatusCode: 403,
```

Comment:
should it be 404?

Reply:
No, I saw real backend returning 403.

The [docs](https://docs.databricks.com/api/workspace/jobs/reset) reserve 404 for something else:

> 403 PERMISSION_DENIED Caller does not have permission to execute the specified operation.
> 404 FEATURE_DISABLED If a given user/entity is trying to use a feature which has been disabled.